addons/airbrake/src/main/scala/com/spingo/op_rabbit/AirbrakeLogger.scala

```diff
@@ -42,30 +46,36 @@ class AirbrakeLogger(appName: String, airbrakeKey: String, environment: String)
     val notice = new AirbrakeNoticeBuilder(airbrakeKey, exception, environment) {
       setRequest(s"consumer://$appName/$name", "consume") // it's a faux URL, but reduces weirdness when clicking in airbrake webapp
 
-      val headerProperties: Map[String, String] = Option(properties.getHeaders) map { _.map { case (k,v) => (s"HEADER:$k", v.toString) }.toMap } getOrElse Map.empty
+      val headerProperties: Map[String, String] = Option(properties.getHeaders.asScala) map { _.map { case (k,v) => (s"HEADER:$k", v.toString) }.toMap } getOrElse Map.empty
 
       session(
-        Map("consumerTag" -> consumerTag))
+        asJavaMap(Map("consumerTag" -> consumerTag)))
 
       request(
-        Map(
+        asJavaMap(Map(
           "body"        -> bodyAsString(body, properties),
           "deliveryTag" -> envelope.getDeliveryTag.toString,
           "redeliver"   -> envelope.isRedeliver.toString,
           "exchange"    -> envelope.getExchange,
-          "routingKey"  -> envelope.getRoutingKey) ++ headerProperties)
+          "routingKey"  -> envelope.getRoutingKey) ++ headerProperties))
 
       environment(
-        Map(
+        asJavaMap(Map(
           "host" -> java.net.InetAddress.getLocalHost.getHostName,
-          "consumer" -> name))
+          "consumer" -> name)))
 
       projectRoot(appName)
     }
     new AirbrakeNotifier().notify(notice.newNotice())
   } catch {
     case e: Throwable => log.error("Unable to send airbrake notification for error", e)
   }
+
+  private def asJavaMap(map: Map[String, String]): java.util.Map[String, Object] = {
```

Comment:
Why is this method needed? Isn't it sufficient just to call "Map().asJava" ?

Reply:
asJava was returning Map<String, String> when the Airbrake APIs require Map<String, Object> so I thought it was better to create a function to produce maps of the required type